zerver/tests/test_user_groups.py

```diff
@@ -352,39 +352,73 @@ def test_update_members_of_user_group(self) -> None:
         othello = self.example_user("othello")
         add = [othello.id]
         params = {"add": orjson.dumps(add).decode()}
-        result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
-        self.assert_json_success(result)
+
+        events: List[Mapping[str, Any]] = []
+
+        with self.tornado_redirected_to_list(events, expected_num_events=2):
+            result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
+            self.assert_json_success(result)
+
         self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 2)
         members = get_direct_memberships_of_users(user_group, [hamlet, othello])
         self.assert_length(members, 2)
 
+        # A notification message is sent for adding to user group
+        msg_event = [e for e in events if e["event"]["type"] == "message"]
+        self.assert_length(msg_event, 1)
```

Comment:
Can we also check the message content probably from the message object like we do for notifications sent on changing stream permissions?

Reply:
Yes ! Now the test checks message content also.